cradle-admin-tool-http/src/main/java/com/exactpro/th2/cradle/adm/http/PageManager.java

```diff
@@ -117,8 +148,48 @@ public void run() {
             try {
                 autoPageInfo.setBookInfo(checkBook(autoPageInfo.getBookInfo(), autoPageInfo.getAutoPageConfiguration()));
             } catch (Exception e) {
-                logger.error("Exception processing book {}", bookName, e);
+                LOGGER.error("Exception processing book {}", bookName, e);
             }
         });
     }
+
+    private static BookInfo getOrCreateBook(@NotNull CradleStorage storage, String bookName, boolean autoBook) {
+        try {
+            BookInfo storageBookInfo = storage.refreshBook(bookName);
+
+            if (storageBookInfo != null) {
+                return storageBookInfo;
+            }
+
+            if (!autoBook) {
+                throw new IllegalStateException("Storage doesn't contain the '" + bookName + "' book, auto book: false");
```

Comment:
I think we should make the exception more informative
```suggestion
                throw new IllegalStateException("Storage doesn't contain the '" + bookName + "' book. Create book manually or enable auto-book functionality in configuration");
```